torch_xla/torch_xla.py

```diff
@@ -134,28 +166,35 @@ def _compile():
     saved_allow_execution = torch_xla._XLAC._get_allow_execution()
     saved_current_graph_name = torch_xla._XLAC._get_current_graph_name()
     torch_xla._XLAC._set_use_eager_mode(False)
-    if name != None:
+    if name is not None:
       torch_xla._XLAC._set_current_graph_name(name + '_clear_pending')
     # Clear pending operations
     _clear_pending_ops_before_compile()
 
-    if name != None:
+    if name is not None:
       torch_xla._XLAC._set_current_graph_name(name)
 
     # if full_graph sets to true execution can not happen before the sync below
     torch_xla._XLAC._set_allow_execution(not full_graph)
 
+    if num_different_graphs_allowed is not None:
+      torch_xla._XLAC._dynamic_shape_detector_set_max_num_different_graphs_allowed(
+          num_different_graphs_allowed)
+      torch_xla._XLAC._dynamic_shape_detector_start_session(current_id)
```

Comment:
I felt like it is better to regsiter the session with `num_different_graphs_allowed ` outside of the `_compile ` and in here we just need to start the session. We can do that in the follow up